src/crypto/srtp.rs

```diff
@@ -0,0 +1,251 @@
+use std::fmt;
+
+use self::aead_aes_128_gcm::AeadKey;
+use self::aes_128_cm_sha1_80::AesKey;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SrtpProfile {
+    #[cfg(feature = "_internal_test_exports")]
+    PassThrough,
+    Aes128CmSha1_80,
+    AeadAes128Gcm,
+}
+
+impl SrtpProfile {
+    // All the profiles we support, ordered from most preferred to least.
+    pub(crate) const ALL: &'static [SrtpProfile] =
+        &[SrtpProfile::AeadAes128Gcm, SrtpProfile::Aes128CmSha1_80];
+
+    /// The length of keying material to extract from the DTLS session in bytes.
+    #[rustfmt::skip]
+    pub(crate) fn keying_material_len(&self) -> usize {
+        match self {
+            #[cfg(feature = "_internal_test_exports")]
+            SrtpProfile::PassThrough => 0,
+             // MASTER_KEY_LEN * 2 + MASTER_SALT * 2
+             // TODO: This is a duplication of info that is held in srtp.rs, because we
+             // don't want a dependency in that direction.
+            SrtpProfile::Aes128CmSha1_80 => 16 * 2 + 14 * 2,
+            SrtpProfile::AeadAes128Gcm   => 16 * 2 + 12 * 2,
+        }
+    }
+}
+
+#[allow(unused)]
+
+pub fn new_aes_128_cm_sha1_80(
+    key: AesKey,
+    encrypt: bool,
+) -> Box<dyn aes_128_cm_sha1_80::CipherCtx> {
+    #[cfg(feature = "openssl")]
+    {
+        let ctx = super::ossl::OsslSrtpCryptoImpl::new_aes_128_cm_sha1_80(key, encrypt);
+        Box::new(ctx)
+    }
+    #[cfg(not(feature = "openssl"))]
+    {
+        panic!("No SRTP implementation. Enable openssl feature");
+    }
+}
+
+#[allow(unused)]
+
+pub fn new_aead_aes_128_gcm(key: AeadKey, encrypt: bool) -> Box<dyn aead_aes_128_gcm::CipherCtx> {
+    #[cfg(feature = "openssl")]
+    {
+        let ctx = super::ossl::OsslSrtpCryptoImpl::new_aead_aes_128_gcm(key, encrypt);
+        Box::new(ctx)
+    }
+    #[cfg(not(feature = "openssl"))]
+    {
+        panic!("No SRTP implementation. Enable openssl feature");
+    }
+}
+
+#[allow(unused)]
+
+pub fn srtp_aes_128_ecb_round(key: &[u8], input: &[u8], output: &mut [u8]) {
+    #[cfg(feature = "openssl")]
+    {
+        super::ossl::OsslSrtpCryptoImpl::srtp_aes_128_ecb_round(key, input, output)
+    }
+    #[cfg(not(feature = "openssl"))]
+    {
+        panic!("No SRTP implementation. Enable openssl feature");
+    }
+}
```

Comment:
I am surprised to see this here when you've (correctly) advocated for additive features previously. How is this going to evolve in the future?

Reply:
It's a TBD. Passing the openssl config RtcConfig -> SrtpContext felt a bit too far for me to think about today. But you're right, I should leave a comment here.